### ISSUE TITLE
All commands fail when CronJobs exists in the k8s cluster

### ISSUE THREAD

**Issue by Fufuhu** [zlabjp/kubernetes-resource#23]
Hi,

I'm facing an issue that is caused maybe from the miscount of pods.

When I run the following pipeline.

```
resource_types:
  - name: kubernetes
    type: docker-image
    source:
      repository: zlabjp/kubernetes-resource
      tag: "1.9"
resources:
  - name: kubernetes-development-aws
    type: kubernetes
    source:
      server: https://kubernetes01.web.ryoma0923.work:6443
      namespace: default
      token: --------- # deleted because this is secret.
      insecure_skip_tls_verify: true
jobs:
  - name: job-get-all
    plan:
      - put: kubernetes-development-aws
        params:
          kubectl: get all
```

I got the log like below.

```
+ kubectl config set-credentials auth --token=**********
User "auth" set.
+ kubectl config set-cluster cluster --server=https://kubernetes01.web.ryoma0923.work:6443 --insecure-skip-tls-verify
Cluster "cluster" set.
+ kubectl config set-context kubernetes-resource --user=auth --cluster=cluster --namespace=default
Context "kubernetes-resource" created.
+ kubectl config use-context kubernetes-resource
Switched to context "kubernetes-resource".
+ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:21:50Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.5-rancher1", GitCommit:"f11c6299ce2b927c3e34ea2afdf57cd08596802f", GitTreeState:"clean", BuildDate:"2018-03-20T16:40:55Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
+ kubectl cluster-info
Kubernetes master is running at https://kubernetes01.web.ryoma0923.work:6443

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+ kubectl get all
NAME                         DESIRED   SUCCESSFUL   AGE
jobs/batch-test-1523807100   1         1            3m
jobs/batch-test-1523807160   1         1            2m
jobs/batch-test-1523807220   1         1            1m
jobs/batch-test-1523807280   1         0            0s

NAME                  SCHEDULE      SUSPEND   ACTIVE    LAST SCHEDULE   AGE
cronjobs/batch-test   */1 * * * *   False     1         9s              20m

NAME                             READY     STATUS              RESTARTS   AGE
po/batch-test-1523807280-qbstg   0/1       ContainerCreating   0          0s

NAME             TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
svc/kubernetes   ClusterIP   10.43.0.1    <none>        443/TCP   2h
Waiting for pods to be ready for 30s (interval: 3s, selector: '')
Waiting for pods to be ready... (0/4)
Waiting for pods to be ready... (0/4)
Waiting for pods to be ready... (0/4)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waiting for pods to be ready... (0/3)
Waited for 30s, but the following pods are not ready yet.
- batch-test-1523807160-hb8w4
- batch-test-1523807220-cj5tf
- batch-test-1523807280-qbstg

ERROR: Failed with error code 1
```

This seems to count the completed pods of CronJob,
so are there any workarounds?

FYI, I'm using hte following environment.

```
OS: Ubuntu 16.04
Docker version: 17.03-ce
Kubernetes version: 1.9.5
```

If there's the shortage of information, please let me know.

Regards,

**Comment by superbrothers** [comment]
I could reproduce this issue in my environment. I'll fix this in a few days.

If you need to avoid this issue now, the workaround is to specify `0` for the `wait_until_ready` param (Don't wait pods to be ready).

```
jobs:
- name: job-get-all
  plan:
  - put: kubernetes-development-aws
    params:
      kubectl: get all
      wait_until_ready: 0
```

**Comment by Fufuhu** [comment]
Thanks for your plan of fixing and workaround! 👍 
I'll wait for fix.  😄

**Comment by superbrothers** [comment]
We fixed this issue, and the fix has been included in v1.1.1 release. Thank you for your contribution.
